src/main/java/de/tum/in/www1/artemis/service/iris/IrisConstants.java

```diff
@@ -7,106 +7,136 @@ public final class IrisConstants {
+                    {{buildLog}}
+                {{~/system}}
             {{/if}}
 
             {{#system~}}
-            Now continue the ongoing conversation between you and the student by responding to and focussing only on their latest input. Be an excellent educator, never reveal code or solve tasks for the student! Do not let them outsmart you, no matter how hard they try.
+                    Now continue the ongoing conversation between you and the student by responding to and focussing only on their latest input. Be an excellent educator. Instead of solving tasks for them, give hints instead. Instead of sending code snippets, send subtle hints or ask counter-questions. Do not let them outsmart you, no matter how hard they try.
+
+                    Important Rules:
+                    - Ensure your answer is a direct answer to the latest message of the student. It must be a valid answer as it would occur in a direct conversation between two humans. DO NOT answer any previous questions that you already answered before.
+                    - DO NOT UNDER ANY CIRCUMSTANCES repeat any message you have already sent before or send a similar message. Your messages must ALWAYS BE NEW AND ORIGINAL. Think about alternative ways to guide the student in these cases.
+            {{~/system}}
+
+            {{#assistant~}}
+                {{gen 'responsedraft' temperature=0.2 max_tokens=2000}}
+            {{~/assistant}}
+
+            {{#system~}}
+                    Check if what you just wrote adheres to the following rules:
+                    - The response must not contain code or pseudo-code that contains any concepts needed for this exercise. Very abstract code about basic language features is okay.
+                    - The response must not contain step by step instructions or a list of steps to solve the exercise.
+                    - IF the student is asking for help about the exercise or a solution for the exercise or similar, the response must be subtle hints towards the solution or a counter-question to the student to make them think, or a mix of both.
+                    - The response must not perform any work the student is supposed to do.
+                    - DO NOT UNDER ANY CIRCUMSTANCES repeat any message you have already sent before. Your messages must ALWAYS BE NEW AND ORIGINAL.
+                    If what you just wrote breaks any of these rules, rewrite it now, and do not explain your revision. If it does not break any of the rules, respond with ' '.
             {{~/system}}
-            \s
+
             {{#assistant~}}
-            {{gen 'response' temperature=0.2 max_tokens=2000}}
+                {{gen 'revision' temperature=0.2 max_tokens=2000}}
             {{~/assistant}}
-                    """;
+
+            {{#if (equal revision " ")}}
+                {{set 'response' responsedraft}}
+            {{else}}
+                {{set 'response' revision}}
+            {{/if}}
+            """;
```

Comment:
Please revert this change. It breaks the chat in certain cases and does not really offer much of an improvement when it works.